core/src/Groupper.ts

```diff
@@ -10,11 +10,34 @@ import { getTabsterOnElement } from './Instance';
 import { Keys } from './Keys';
 import { RootAPI } from './Root';
 import * as Types from './Types';
-import { getLastChild, TabsterPart, WeakHTMLElement } from './Utils';
+import { DummyInput, DummyInputManager, getLastChild, TabsterPart, WeakHTMLElement } from './Utils';
+
+class GroupperDummyManager extends DummyInputManager {
+    private _tabster: Types.TabsterCore;
+
+    constructor(element: WeakHTMLElement, tabster: Types.TabsterCore) {
+        super(tabster, element);
+        this._tabster = tabster;
+        this.firstDummy.onFocusIn = this._onFocusDummyInput;
+        this.lastDummy.onFocusIn = this._onFocusDummyInput;
+    }
+
+    private _onFocusDummyInput = (dummyInput: DummyInput) => {
+        const container = this._element.get();
+        if (container && !dummyInput.shouldMoveOut) {
+            if (dummyInput.isFirst) {
+                this._tabster.focusedElement.focusFirst({ container });
+            } else {
+                this._tabster.focusedElement.focusLast({ container });
+            }
+        }
+    }
+}
 
 export class Groupper extends TabsterPart<Types.GroupperProps> implements Types.Groupper {
     private _shouldTabInside = false;
     private _first: WeakHTMLElement | undefined;
+    public _dummyManager?: GroupperDummyManager;
```

Comment:
`public` is not used explicitly over the codebase, perhaps we should keep it consistent here and in other changes from this PR (and have a prettier rule to use it or not once we have it).

Reply:
removed usages of `public` from this PR